aiida/orm/utils/repository.py

```diff
@@ -175,18 +175,35 @@ def put_object_from_file(self, path, key, mode='w', encoding='utf8', force=False
         :param path: absolute path of file whose contents to copy to the repository
         :param key: fully qualified identifier for the object within the repository
         :param mode: the file mode with which the object will be written
+            Deprecated: will be removed in `v1.1.0`
         :param encoding: the file encoding with which the object will be written
+            Deprecated: will be removed in `v1.1.0`
         :param force: boolean, if True, will skip the mutability check
         :raises aiida.common.ModificationNotAllowed: if repository is immutable and `force=False`
         """
-        # pylint: disable=unused-argument
+        # pylint: disable=unused-argument,no-member
+        import warnings
+        from aiida.common.warnings import AiidaDeprecationWarning
+
+        # Note that the defaults of `mode` and `encoding` had to be change to `None` from `w` and `utf-8` resptively, in
+        # order to detect when they were being passed such that the deprecation warning can be emitted. The defaults did
+        # not make sense and so ignoring them is justified, since the side-effect of this function, a file being copied,
+        # will continue working the same.
+        if mode is not None:
+            warnings.warn('the `mode` argument is deprecated and will be removed in `v1.1.0`', AiidaDeprecationWarning)
+
+        if encoding is not None:
+            warnings.warn(
+                'the `encoding` argument is deprecated and will be removed in `v1.1.0`', AiidaDeprecationWarning
+            )
+
         if not force:
             self.validate_mutability()
 
         self.validate_object_key(key)
 
         with io.open(path, mode='rb') as handle:
-            self.put_object_from_filelike(handle, key, mode='wb')
+            self.put_object_from_filelike(handle, key, mode='wb', encoding=None)
```

Comment:
Just out of curiosity: why add this encoding argument here?

Reply:
because the default encoding is `utf-8` which does not make sense and if the lower lying code does the wrong thing, it will crash. It so happens that there is a line in `Folder.create_from_filelike` that sets `encoding` to `None` if `b` is in the mode, but again, it is better to do this here.